### ISSUE TITLE
identifying nuclear/cytoplasmic transcripts with segger

### ISSUE THREAD

**Issue by jpark27** [dpeerlab/segger#69]
Dear segger team,

Hello, all! Hope you been well.

I have been testing some tools with compartmentalised counts (e.g., nuclear/cytoplasmic) from Xenium 5K prime data. With segger, on final output (`segmentation.zarr`), is there a way to add layers with transcripts having "nuclear" and  "cytoplasmic" counts separately?

To do that, do I have to re-run segger? Anyhow, couldnt find any option from segger segment --help:

Any help would be super appreciated!
J

╭─ Parameters ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ DEBUG --debug --no-debug  Whether to save additional debug information (trainer, predictions). [default: none]                                     │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ I/O ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to file inputs/outputs.                                                                                                                    │
│                                                                                                                                                    │
│ *  INPUT-DIRECTORY --input-directory -i           Path to the standardized IST dataset directory. [required]                                       │
│ *  OUTPUT-DIRECTORY --output-directory -o         Path to write outputs. [required]                                                                │
│    SAVE-ANNDATA --save-anndata --no-save-anndata  [default: True]                                                                                  │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Node Representation ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to transcript and cell node representations.                                                                                               │
│                                                                                                                                                    │
│ NODE-REPRESENTATION-DIM --node-representation-dim      Number of dimensions used to represent each node type. [default: 128]                       │
│ CELLS-REPRESENTATION --cells-representation            Feature representation used for cell embeddings. [choices: pca, morphology] [default: pca]  │
│ CELLS-MIN-COUNTS --cells-min-counts                    Minimum transcript count threshold per cell. [default: 10]                                  │
│ CELLS-CLUSTERS-N-NEIGHBORS                             Number of neighbors for cell clustering. [default: 10]                                      │
│   --cells-clusters-n-neighbors                                                                                                                     │
│ CELLS-CLUSTERS-RESOLUTION --cells-clusters-resolution  Resolution parameter for cell clustering. [default: 2.0]                                    │
│ GENES-CLUSTERS-N-NEIGHBORS                             Number of neighbors for gene clustering. [default: 5]                                       │
│   --genes-clusters-n-neighbors                                                                                                                     │
│ GENES-CLUSTERS-RESOLUTION --genes-clusters-resolution  Resolution parameter for gene clustering. [default: 2.0]                                    │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Transcript-Transcript Graph ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to transcript-transcript graph parameters.                                                                                                 │
│                                                                                                                                                    │
│ TRANSCRIPTS-MAX-K --transcripts-max-k        Maximum number of edges per transcript in the local graph. [default: 5]                               │
│ TRANSCRIPTS-MAX-DIST --transcripts-max-dist  Maximum edge distance for transcript graph construction. [default: 5.0]                               │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Segmentation (Prediction) Graph ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to segmentation prediction graph parameters.                                                                                               │
│                                                                                                                                                    │
│ PREDICTION-MODE --prediction-mode    Graph structure used during prediction. [choices: nucleus, cell, uniform] [default: cell]                     │
│ PREDICTION-MAX-K --prediction-max-k  Maximum number of edges per transcript for prediction graphs. [default: 3]                                    │
│ PREDICTION-EXPANSION-RATIO           [default: 0.05]                                                                                               │
│   --prediction-expansion-ratio                                                                                                                     │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Tiling ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to tiling parameters.                                                                                                                      │
│                                                                                                                                                    │
│ TILING-MARGIN-TRAINING --tiling-margin-training      Margin width (in µm) added to tiles during training. [default: 20.0]                          │
│ TILING-MARGIN-PREDICTION --tiling-margin-prediction  Margin width (in µm) added to tiles during prediction. [default: 20.0]                        │
│ MAX-NODES-PER-TILE --max-nodes-per-tile              Maximum number of nodes per tile for adaptive tiling. [default: 50000]                        │
│ MAX-EDGES-PER-BATCH --max-edges-per-batch            Maximum number of edges per batch in the DataLoader. [default: 1000000]                       │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Model ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to model architecture and training parameters.                                                                                             │
│                                                                                                                                                    │
│ N-EPOCHS --n-epochs                                    Number of training epochs. [default: 20]                                                    │
│ N-MID-LAYERS --n-mid-layers                            [default: 2]                                                                                │
│ N-HEADS --n-heads                                      [default: 2]                                                                                │
│ HIDDEN-CHANNELS --hidden-channels                      [default: 64]                                                                               │
│ OUT-CHANNELS --out-channels                            [default: 64]                                                                               │
│ LEARNING-RATE --learning-rate                          [default: 0.001]                                                                            │
│ USE-POSITIONAL-EMBEDDINGS --use-positional-embeddings  [default: True]                                                                             │
│   --no-use-positional-embeddings                                                                                                                   │
│ NORMALIZE-EMBEDDINGS --normalize-embeddings            [default: True]                                                                             │
│   --no-normalize-embeddings                                                                                                                        │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Loss ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Related to loss function parameters.                                                                                                               │
│                                                                                                                                                    │
│ SEGMENTATION-LOSS --segmentation-loss              [choices: triplet, bce][default: triplet]                                                       │
│ TRANSCRIPTS-MARGIN --transcripts-margin            [default: 0.3]                                                                                  │
│ SEGMENTATION-MARGIN --segmentation-margin          [default: 0.4]                                                                                  │
│ TRANSCRIPTS-LOSS-WEIGHT-START                      [default: 1.0]                                                                                  │
│   --transcripts-loss-weight-start                                                                                                                  │
│ TRANSCRIPTS-LOSS-WEIGHT-END                        [default: 1.0]                                                                                  │
│   --transcripts-loss-weight-end                                                                                                                    │
│ CELLS-LOSS-WEIGHT-START --cells-loss-weight-start  [default: 1.0]                                                                                  │
│ CELLS-LOSS-WEIGHT-END --cells-loss-weight-end      [default: 1.0]                                                                                  │
│ SEGMENTATION-LOSS-WEIGHT-START                     [default: 0.0]                                                                                  │
│   --segmentation-loss-weight-start                                                                                                                 │
│ SEGMENTATION-LOSS-WEIGHT-END                       [default: 0.5]                                                                                  │
│   --segmentation-loss-weight-end                                                                                                                   │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

**Comment by Tobiaspk** [comment]
Hi @jpark27 , Xenium's transcripts file already includes a nucleus-assignment column. Easiest would probably be simply joining that column to the segger outputs. Any transcript that is not in nucleus, but assigned by segger, would be in the cytoplasm. Does that answer your question?

**Comment by Tobiaspk** [comment]
Hi @jpark27 , following-up if that solved your issue.

**Comment by jpark27** [comment]
> Hi [@jpark27](https://github.com/jpark27) , following-up if that solved your issue.

Dear @Tobiaspk! Oh, yes! I resolved issue by pulling out using overlaps_nucleus flag from parquet file. Many thanks,

Jun
